### DISCUSSION TITLE
Release v1.6.0 alpha

### DISCUSSION THREAD

**Discussion by RevZero** [X2CommunityCore/xcom2-launcher#385] [-Releases-]
**Download: [XCOM2_AlternativeModLauncher_1.6.0-alpha.zip](https://github.com/X2CommunityCore/xcom2-launcher/releases/download/v1.6.0-alpha/XCOM2_AlternativeModLauncher_1.6.0-alpha.zip)**
> **Note that this is a alpha-release and may contain bugs. Consider waiting for the final release to get a more polished experience.**

## Preface

Hi everyone! 

I finally found the time to properly test v1.6. This release includes some extensive changes, mainly aiming to improve performance when initializing/updating mods. There are also some usability improvements and fixes.

**Attention:** If you manually edited mod descriptions (to makes notes), be aware, that all changes will be overwritten. Until now, mod descriptions were actually never updated.

Thanks to @remcoros for contributing.

I am always grateful for a [coffee donation ☕](https://www.buymeacoffee.com/RevZero).

## Infos

Refer to the [installation instructions](https://github.com/X2CommunityCore/xcom2-launcher/wiki/Installation) about how to install or update AML.

- Check out [the Wiki](https://github.com/X2CommunityCore/xcom2-launcher/wiki) for some useful tips.
- For questions or suggestions you are welcome to join our [Discord channel](https://discord.gg/QHSVGRn) or start a [Discussion](https://github.com/X2CommunityCore/xcom2-launcher/discussions).
- If AML is crashing or otherwise not working as expected, you can [report a bug here](https://github.com/X2CommunityCore/xcom2-launcher/issues/new/choose).

### New features

- Added new setting: Toggle if mods are automatically updated on startup (enabled by default). This is for you @Iridar :)
- Huge performance increase when updating mods. @remcoros figured out how to fetch the mod dependencies while requesting the mod details. Previously this required a whole bunch of additional requests. 
- Increased performance when there are lots of dependencies which are currently not installed. @RevZero 
- Added support to make notes for each mod. Notes can be edited from the new note tab in mod info panel or using the note column in the mod list view (column is hidden by default). The mod description can no longer be edited. @RevZero 

### Usability

- All dialogs can now be closed by pressing the escape key. Fixes #332. @RevZero 
- Enter key can now be used to close/confirm dialogs. @remcoros 
- The initial WelcomeDialog can now be properly "canceled", effectively closing AML. Until now, it was possible to skip the dialog by pressing ALT+F4 and AML continued with default values. @RevZero 

### Misc improvements

- Tweaks to mod list being non responsive and prevent unnecessarily refreshing/redrawing it. @remcoros 
- Consider XComEngine.ini files from Config sub-folders when reading MCOs. #348 @remcoros 
- Remove unused "Compare" button in ConfigDiff dialog.

### Fixes

- Mod update process not finishing. #317 @remcoros 
- AML crashing when starting the game. #289 #268. @remcoros 
- Mod descriptions not updating. #280 @RevZero 

### Technical

- Query mod information for uninstalled mod dependencies in batches instead of executing a query for each mod. @RevZero 
- Overall improvements to Steam API integration. #345 @remcoros 
- Improved asynchronous processing to keep app responsive. @remcoros 
- Load mod dependencies as part of the GetDetails batch requests. @remcoros 
- Dispose all dialogs properly. @RevZero 
- Only update dependencies for mods that were actually updated instead of all. @RevZero 

<hr /><em>This discussion was created from the release <a href='https://github.com/X2CommunityCore/xcom2-launcher/releases/tag/'>Release v1.6.0 alpha</a>.</em>

**Comment by Dragon-32** [comment]
Mentioned this on Discord: Ordering of Categories doesn't seem to be obeyed, and rejiggering their order causes them to expand on the main pane but doesn't affect their order.

Despite that don't think I'm going to be going back to 1.5.*, thanks for this.

[Edit: Clicked the column header a couple of times and Category order is what I've set it to. idk]

**Comment by caseyjp11** [comment]
Granted I'm coming from the Linux/Proton side, but I've been using AML for a couple years and it works identically to the windows counterpart, and literally kept me playing the game as I won't use Microsoft's security sieve software for any reason.
After dropping the new files into the AML folder and firing up the game:
The changlog area still shows as blank.
On Linux (proton-ge or official Proton), tooltips crash/freeze the application. (I know you think this is proton's fault, and I tend to agree, but the more info the merrier. )

The mod load times definitely improved. 
The game launches just fine (if I avoid things with tooltips).

**Comment by Tedster59** [comment]
![image](https://github.com/X2CommunityCore/xcom2-launcher/assets/21696818/61fde65b-e8b6-4023-aa68-4959691e48d2)

Dragging columns too short for all the text to fit causes it to wrap when there isn't the proper room for it.
